### ISSUE TITLE
LG TV turns on and then off again

### ISSUE THREAD

**Issue by jcc2323** [JPersson77/LGTVCompanion#159]
When I use sleep on my windows 11 the TV will turn off.
When I press a key on the keyboard to wake the PC it will turn on the TV but within a few sec will turn it off again.
If I use the remote to turn it on again it too soon it might turn off again.

I need to wait like 10 sec or so before using the remote to turn the TV on.

**Comment by jcc2323** [comment]
![lg tv settings](https://github.com/JPersson77/LGTVCompanion/assets/59026547/97a892c0-4637-4e4f-bdc7-b9eca896ef08)
The settings should be correct.
When I manually click turn on and turn off via the software 3.15 it will work as intended.

**Comment by JPersson77** [comment]
Hi @jcc2323,  can you please enable logging and upload it to pastebin along with config.json and copy the links here so I can have a  look. Both are found in c:/programdata/lgtv companion/

**Comment by jcc2323** [comment]
Fri 14:19:00 > ** System resumed from low power state (Automatic).
Fri 14:19:01 > ** System requests displays ON.
Fri 14:19:01 > Device1, spawning Thread_DisplayOn().
Fri 14:19:01 > Device1, repeating WOL broadcast started to MAC: D48660681ABE using IP address: 192.168.1.49
Fri 14:19:01 > Device1, Best route to IP - interface index 7 LUID 1689399632855040 route protocol 2
Fri 14:19:01 > Device1, WARNING! Thread_DisplayOn(): connect: A socket operation was attempted to an unreachable host [system:10065 at D:\a\LGTVCompanion\LGTVCompanion\vcpkg_installed\x64-windows-static\x64-windows-static\include\boost\asio\detail\win_iocp_socket_service.hpp:632:5 in function 'connect']
Fri 14:19:02 > Device1, WARNING! Thread_DisplayOn(): connect: A socket operation was attempted to an unreachable host [system:10065 at D:\a\LGTVCompanion\LGTVCompanion\vcpkg_installed\x64-windows-static\x64-windows-static\include\boost\asio\detail\win_iocp_socket_service.hpp:632:5 in function 'connect']
Fri 14:19:04 > ** System resumed from low power state.
Fri 14:19:07 > -daemon 1 topology state 
Fri 14:19:07 > [IPC Daemon 1] Windows monitor topology was changed. Device1:OFF 
Fri 14:19:07 > Device1, spawning Thread_DisplayOff().
Fri 14:19:07 > Device1, [DEBUG] (SSL) OFF response 1: {"type":"registered","id":"register_0","payload":{"client-key":"0aaf443c68196f4d1c62867e2946c9a3"}}
Fri 14:19:07 > Device1, [DEBUG] (SSL) OFF response 2: {"type":"response","id":44,"payload":{"returnValue":true,"state":"Suspend","processing":"Screen On"}}
Fri 14:19:07 > Device1, power state is: Suspend
Fri 14:19:11 > -daemon 1 topology state 
Fri 14:19:11 > [IPC Daemon 1] Windows monitor topology was changed. Device1:OFF 
Fri 14:19:11 > Device1, spawning Thread_DisplayOff().
Fri 14:19:11 > Device1, [DEBUG] (SSL) OFF response 1: {"type":"registered","id":"register_0","payload":{"client-key":"0aaf443c68196f4d1c62867e2946c9a3"}}
Fri 14:19:11 > Device1, [DEBUG] (SSL) OFF response 2: {"type":"response","id":45,"payload":{"returnValue":true,"state":"Active"}}
Fri 14:19:11 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"0aaf443c68196f4d1c62867e2946c9a3"}}
Fri 14:19:11 > Device1, [DEBUG] (SSL) OFF response 4: {"type":"response","id":46,"payload":{"returnValue":true}}
Fri 14:19:11 > Device1, power state is: OFF.
Fri 14:19:11 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":47,"error":"500 Application error","payload":{"returnValue":false,"errorCode":"-2","errorText":"com.webos.service.tvpower/power is busy"}}
Fri 14:19:11 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":48,"payload":{"returnValue":true,"state":"Active","processing":"Request Power Off Logo"}}
Fri 14:19:11 > Device1, power state is: ON
Fri 14:19:11 > Device1, repeating WOL broadcast ended
Fri 14:19:22 > -daemon 1 topology state Device1 
Fri 14:19:22 > [IPC Daemon 1] Windows monitor topology was changed. Device1:ON 
Fri 14:19:22 > Device1, spawning Thread_DisplayOn().
Fri 14:19:22 > Device1, repeating WOL broadcast started to MAC: D48660681ABE using IP address: 192.168.1.49
Fri 14:19:22 > Device1, Best route to IP - interface index 7 LUID 1689399632855040 route protocol 2
Fri 14:19:23 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"0aaf443c68196f4d1c62867e2946c9a3"}}
Fri 14:19:23 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":50,"error":"500 Application error","payload":{"returnValue":false,"state":"Active","errorCode":"-102","errorText":"The current state must be 'Screen Off'"}}
Fri 14:19:23 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":1,"payload":{"returnValue":true,"state":"Active"}}
Fri 14:19:23 > Device1, power state is: ON
Fri 14:19:23 > Device1, repeating WOL broadcast ended
Fri 14:19:27 > -daemon 1 topology state Device1 
Fri 14:19:27 > [IPC Daemon 1] Windows monitor topology was changed. Device1:ON 
Fri 14:19:27 > Device1, spawning Thread_DisplayOn().
Fri 14:19:27 > Device1, repeating WOL broadcast started to MAC: D48660681ABE using IP address: 192.168.1.49
Fri 14:19:27 > Device1, Best route to IP - interface index 7 LUID 1689399632855040 route protocol 2
Fri 14:19:28 > Device1, [DEBUG] (SSL) ON response 1: {"type":"registered","id":"register_0","payload":{"client-key":"0aaf443c68196f4d1c62867e2946c9a3"}}
Fri 14:19:28 > Device1, [DEBUG] (SSL) ON response 2: {"type":"error","id":3,"error":"500 Application error","payload":{"returnValue":false,"state":"Active","errorCode":"-102","errorText":"The current state must be 'Screen Off'"}}
Fri 14:19:28 > Device1, [DEBUG] (SSL) ON response 3: {"type":"response","id":4,"payload":{"returnValue":true,"state":"Active"}}
Fri 14:19:28 > Device1, power state is: ON
Fri 14:19:29 > Device1, repeating WOL broadcast ended

**Comment by jcc2323** [comment]
I just pressed the CTRL key and it started up and TV turned on and around 1-2 sec it turned off again.  I waited a few sec and manually used the TV remote to turn it on again.

**Comment by JPersson77** [comment]
There are several topology changes in a matter of seconds, i e lines like these: 

Fri 14:19:07 > -daemon 1 topology state
Fri 14:19:07 > [IPC Daemon 1] Windows monitor topology was changed. Device1:OFF

I e the app is detecting that the tv is not part of the current monitor topology and therefore power it off. Basically your system does not behave as anticipated, and it is hard to say why without being in front of the computer. Do you have any other applications running which manages the displays? Turning off the montor topology feature should work as a workaround until the root cause is found.

**Comment by jcc2323** [comment]
I have fancyzones from MS powertoys.  Also I have displayfusion

**Comment by jcc2323** [comment]
Is it possible to have a option to tell it to turn on after [x] seconds?  Cause using the turn on function on the lg companion works.  Kinda of a workaround

**Comment by JPersson77** [comment]
Assuming that it is displayfusion that does somethiing. Can you try disabling/not using displayfusion and see what happens?

**Comment by jcc2323** [comment]
I tried uninstalling most of the software that might be involved but problem still persist.  It happens when I sleep and wake the pc up.  It works fine if I shut down and turn on the PC.  Is there a possible fix?  I've just stopped using it for months now since the LG companion just turns it on and off and I need to manually turn it on again anyways.

Is it possible to just have an option to disable LG companion detecting topology and just mannually send a turn on signal after x seconds?  Cause your software turns it on perfectly;  I just need to have it NOT turn it off again.
